src/ui/components/SecondaryToolbox/react-devtools/ReplayWall.ts

```diff
@@ -67,6 +74,25 @@ export class ReplayWall implements Wall {
 
   setPauseId(pauseId: string) {
     this.pauseId = pauseId;
+    this.pauseIdWaiters.get(pauseId)?.resolve();
+    this.pauseIdWaiters.delete(pauseId);
+  }
+
+  private async waitForPauseId(pauseId: PauseId) {
+    if (this.pauseId === pauseId) {
+      return;
+    }
+
+    const waiter = this.pauseIdWaiters.get(pauseId);
+    if (waiter) {
+      await waiter.promise;
+      return;
+    }
+
+    let resolve!: () => void;
+    const promise = new Promise<void>(_resolve => (resolve = _resolve));
+    this.pauseIdWaiters.set(pauseId, { promise, resolve });
+    await promise;
```

Comment:
I think what you have is good, but [`Promise.withResolvers()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers) is also a nifty (more recent) API for doing this sort of thing:

```js
const { promise, resolve } = Promise.withResolvers();
```

Reply:
TIL, nice!

Reply:
But it's too new, we'd need to add a polyfill for that...

Reply:
Yeah, agreed. Though the polyfill would be very small/simple :) I was really glad to see that get added because it's so common and awkward an operation otherwise.